### PR TITLE
lib: modem_attest_token: update to use new AT APIs

### PR DESCRIPTION
**Pull request by lemrey**
Update to use new AT APIs in libmodem.

### CODE REVIEW

**Comment by lemrey** [comment]
disclaimer: I did no testing on this :) @jayteemo will you please check if it's still working?

**Review by SeppoTakalo — APPROVED** [review]
Looks good

**Comment by jayteemo** [comment]
i quickly tried to test with `modem_attest_token_get_uuids` and got a bus fault 😢 
i can look into it more tomorrow.

File: lib/modem_attest_token/modem_attest_token.c [hunk 1]
```diff
@@ -35,87 +35,28 @@
...
-	}
-
-	/* Parse response to get attestation token and COSE strings */
-	ret = parse_resp(cmd_resp, &attest, &cose);
-	if (ret) {
-		ret = -ENOMSG;
-		goto cleanup;
+	ret = nrf_modem_at_scanf(AT_ATTEST_CMD,
+		"%%ATTESTTOKEN: \"%[^.].%[^\"]\"", &attest, &cose);
+	if (ret < 0) {
```
Comment (jtguggedal) [inline lib/modem_attest_token/modem_attest_token.c:58]: Should this check for exactly 2 parameters found?

File: lib/modem_attest_token/modem_attest_token.c [hunk 2]
```diff
@@ -35,87 +35,28 @@
...
-		return -EBADF;
-	}
 
 	int ret = 0;
-	enum at_cmd_state state;
-	char *cmd_resp = NULL;
-	char *attest = NULL;
-	char *cose = NULL;
+	char attest[64] = {0};
+	char cose[64] = {0};
```
Comment (lemrey) [inline lib/modem_attest_token/modem_attest_token.c:43]: Anybody knows how large can these really be?

Comment (SeppoTakalo) [reply lib/modem_attest_token/modem_attest_token.c:43]: Example seems quite small https://infocenter.nordicsemi.com/topic/ref_at_commands/REF/at_commands/security/attesttoken_set.html

I'm not sure if these are fixed in size, Hasnain probably knows. (I'll try to find his Github account)

Comment (jayteemo) [reply lib/modem_attest_token/modem_attest_token.c:43]: 64 is too small... this was causing my bus fault.
here is an example attestation token:
```
2dn3hQFQUFA2QjI5T4aAxAYSLimWIwNQVT1XNJ-9R96MMoajfNDqSVCMpoIK91tLoQb5CuTh6FxQ.0oRDoQEmoQRBIfZYQI9Q4JPVaEm5a7zNGIqz9AKnvWQ54L5aFStg-2lQvEnVBV5RJkprOTxrV8ZoHwG0VscGnmoAVDChjHVay4JcH3A
```
original code used:
`#define ATTEST_TOKEN_RESP_LEN 200`
for the response buffer

File: lib/modem_attest_token/modem_attest_token.c (hunk 1 above)
Comment (jayteemo) [reply lib/modem_attest_token/modem_attest_token.c:58]: yes, ret should == 2.
also, you could add length specifiers to the nrf_modem_at_scanf command (if supported).

File: lib/modem_attest_token/modem_attest_token.c [hunk 3]
```diff
@@ -35,87 +35,28 @@
...
-		ret = -EBADMSG;
-		goto cleanup;
-	}
-
-	/* Parse response to get attestation token and COSE strings */
-	ret = parse_resp(cmd_resp, &attest, &cose);
-	if (ret) {
-		ret = -ENOMSG;
-		goto cleanup;
+	ret = nrf_modem_at_scanf(AT_ATTEST_CMD,
```
Comment (jayteemo) [inline lib/modem_attest_token/modem_attest_token.c:56]: if ret == 2, then set `ret = 0;` so the function returns properly.

**Review by jayteemo — CHANGES_REQUESTED** [review]

File: lib/modem_attest_token/modem_attest_token.c (hunk 3 above)
Comment (lemrey) [reply lib/modem_attest_token/modem_attest_token.c:57]: done

File: lib/modem_attest_token/modem_attest_token.c (hunk 2 above)
Comment (lemrey) [reply lib/modem_attest_token/modem_attest_token.c]: thanks, have increased both buffers to 128 bytes

File: lib/modem_attest_token/modem_attest_token.c (hunk 1 above)
Comment (lemrey) [reply lib/modem_attest_token/modem_attest_token.c]: thanks, updated

**Review by jayteemo — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**